src/gax-internal/src/prost.rs

```diff
@@ -15,14 +15,46 @@
 //! Helper functions to convert from the well-known types to and from their
 //! Prost versions.
 
-/// Converts from `Self` into `T`.
-pub trait Convert<T>: Sized {
+use std::collections::BTreeMap;
+
+#[derive(Clone, Debug, PartialEq, thiserror::Error)]
+pub enum ConvertError {
+    #[error("enum {0} does not contain an integer value")]
+    EnumNoIntegerValue(&'static str),
+}
+
+type Result<T> = std::result::Result<T, ConvertError>;
+
+/// Converts from `Self` into `T`, where `T` is expected to be a Protobuf-generated type.
+pub trait ToProto<T>: Sized {
+    type Output;
+    fn to_proto(self) -> Result<Self::Output>;
```

Comment:
Good call. It is easier to keep track of what is a proto vs. what is our type with the new names.